keps/sig-storage/1682-csi-driver-skip-permission/README.md

```diff
@@ -0,0 +1,227 @@
+for CSIDrivers to check the new field, and skip volume ownership modifications if it is found to be
+`false`.
+
+When defining a `CSIDriver`, we propose that `CSIDriver.Spec` be expanded to include a new field entitled 
+`SupportsFSGroup` which can have following possible values:
+
+ - `OnlyRWO` --> Current behavior. Attempt to modify the volume ownership and permissions to the defined `fsGroup` when the volume is 
+ mounted if accessModes is RWO.
+ - `Never` --> New behavior. Attach the volume without attempting to modify volume ownership or permissions.
+ - `Always` --> New behavior. Always attempt to apply the defined fsGroup to modify volume ownership and permissions.
+
+```go
+type CSIDriverFSGroupChangePolicy string
+
+const(
+    OnlyRWO CSIDriverFSGroupChangePolicy = "OnlyRWO"
+    Always CSIDriverFSGroupChangePolicy = "Always"
+    Never CSIDriverFSGroupChangePolicy = "Never"
+)
+
+type CSIDriverSpec struct {
+    // SupportsFSGroup ← new field
+    // Defines if the underlying volume supports changing ownership and 
+    // permission of the volume before being mounted. 
+    // If set to Always, SupportsFSGroup indicates that 
+    // the underlying volume supports volume ownership and permission
+    // changes, and the filesystem will always be modified to match the defined
+    // fsGroup.
+    // If set to Never, then the volume will be mounted without modifying
+    // the volume's ownership or permissions.
+    // Defaults to OnlyRWO, which results in the volume being examined
+    // and the volume ownership and permissions attempting to be updated
+    // if fsGroup is defined and accessModes is RWO.
+    // + optional
+    SupportsFSGroup *CSIDriverFSGroupChangePolicy
+}
+```
+### Test Plan
+
+A test plan will include the following tests:
+
+* Basic tests including a permutation of the following values:
+  - CSIDriver.Spec.SupportsFSGroup  (`nil`/`true`/`false`)
+  - PersistentVolumeClaimStatus.FSGroup (matching, non-matching)
+* E2E tests
+
+### Graduation Criteria
+
+* Alpha in 1.19 provided all tests are passing.
+* All functionality is guarded by a new alpha `CSIVolumeSupportFSGroup` feature gate.
+
+* Beta in 1.20 with design validated by at least two customer deployments
+  (non-production), with discussions in SIG-Storage regarding success of
+  deployments. 
+* The `CSIVolumeSupportFSGroup` feature gate will graduate to beta.
+
+
+* GA in 1.21, with E2E tests in place tagged with feature Storage.
+* The `CSIVolumeSupportFSGroup` feature gate will graduate to GA.
+
+[issues]: https://github.com/kubernetes/enhancements/issues/1682
+
+## Production Readiness Review Questionnaire
+
+### Feature enablement and rollback
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: CSIVolumeSupportFSGroup 
+    - Components depending on the feature gate: CSIDrivers
```

Comment:
this is referring to kubernetes components, like apiserver, kubelet

Reply:
I don't believe there will be any kubernetes components that depend on this, so I've updated it to be None.